pkg/controller/master/backup/restore.go

```diff
@@ -513,54 +560,160 @@ func (h *RestoreHandler) updateOwnerRefAndTargetUID(vmRestore *harvesterv1.Virtu
 }
 
 // createRestoredPVC helps to create new PVC from CSI volumeSnapshot
-func (h *RestoreHandler) createRestoredPVC(vmRestore *harvesterv1.VirtualMachineRestore,
-	volumeBackup harvesterv1.VolumeBackup, volumeRestore harvesterv1.VolumeRestore) error {
+func (h *RestoreHandler) createRestoredPVC(
+	vmRestore *harvesterv1.VirtualMachineRestore,
+	volumeBackup harvesterv1.VolumeBackup,
+	volumeRestore harvesterv1.VolumeRestore,
+) error {
+	if volumeBackup.Name == nil {
+		return fmt.Errorf("missing VolumeSnapshot name")
+	}
+
+	dataSourceName := *volumeBackup.Name
+	if vmRestore.Namespace != volumeBackup.PersistentVolumeClaim.ObjectMeta.Namespace {
+		// create volumesnapshot if namespace is different
+		volumeSnapshot, err := h.getOrCreateVolumeSnapshot(vmRestore, volumeBackup)
+		if err != nil {
+			return err
+		}
+		dataSourceName = volumeSnapshot.Name
+	}
+
+	annotations := map[string]string{}
+	for key, value := range volumeBackup.PersistentVolumeClaim.ObjectMeta.Annotations {
+		needSkip := false
+		for _, prefix := range restoreAnnotationsToDelete {
```

Comment:
Out of curious. What is the purpose of `restoreAnnotationsToDelete`.

Reply:
Not sure why we need to delete some annotations. I just try to refactor how we build annotations map structure. It comes from https://github.com/harvester/harvester/pull/530.